pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -290,5 +290,6 @@ func legacyDeviceSetPVCID(deviceSetName string, setIndex int) string {
 // It includes the pvcTemplateName in it
 func deviceSetPVCID(deviceSetName, pvcTemplateName string, setIndex int) string {
 	cleanName := strings.Replace(pvcTemplateName, " ", "-", -1)
+	deviceSetName = strings.Replace(deviceSetName, ".", "-", -1)
```

Comment:
Any reason why not to allow `.` in the pvc name? It is part of the valid naming convention - https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names

Reply:
Oh of course I should have provided the observed error message. Now the original PR message is updated with the error:
```
2022-12-12 15:08:01.120368 E | op-osd: failed to run osd provisioning job for PVC "name.with.dots-data-02rnf7": 
Job.batch "rook-ceph-osd-prepare-name.with.dots-data-02rnf7" is invalid: [spec.template.spec.volumes[7].name: Invalid value: "name.with.dots-data-02rnf7": 
a lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', 
regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?'), 
spec.template.spec.volumes[8].name: Invalid value: "name.with.dots-data-02rnf7-bridge": 
a lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?'), 
spec.template.spec.containers[0].volumeMounts[7].name: Not found: "name.with.dots-data-02rnf7-bridge", spec.template.spec.initContainers[1].volumeMounts[0].name: Not found: "name.with.dots-data-02rnf7-bridge", spec.template.spec.initContainers[1].volumeDevices[0].name: Not found: "name.with.dots-data-02rnf7"]
```

So it's not in the pvc name where the problem lies, but in the fact that we use the pvc name as the volume name.

Reply:
Got it.  And since the name with dots had never worked before, there won't be scenario where we already have pvcs  like `name.with.dots-data-02rnf7-bridge` in any existing cluster. Is that correct?

Reply:
Yes, correct